packages/flutter/lib/src/gestures/converter.dart

```diff
@@ -32,6 +32,25 @@ class _PointerState {
 
   Offset lastPosition;
 
+  Offset deltaTo(Offset to) {
+    if (to == null) { // [PointerRemovedEvent]
+      return Offset.zero;
+    }
+    return to - lastPosition;
+  }
+
+  /// Update state with the latest event.
+  PointerEvent record(PointerEvent event) {
+    print('${event.pointer} ${event.buttons}');
+    lastPosition = event.position;
+    if (event.down && !_down) {
+      setDown();
+    } else if (!event.down && _down) {
+      setUp();
+    }
+    return event;
+  }
```

Comment:
what is the purpose of the record() method? It looks like it's redundant with existing calls to setDown and setUp

Reply:
Currently we rely on explicit statements in `expand` to update state. Using a `record` makes sure that the state of an accepted event always updates its states.

Reply:
Existing calls of `setUp` `setDown` and setting `lastPosition` have been removed.

Reply:
i'm not really convinced this is better. It makes it easier to make mistakes: if we ever make a change where we shouldn't be changing the state, we won't notice, because this code will be making sure everything keeps working anyway.

I'd much rather rely on explicit statements and liberal asserts.

This approach (with record) is also slower, because it means that we run this code even when we don't have to. It's faster to only change the state when we have to.